ztrouy/cpyou
# Feature - Build Compatibility Checking

### Purpose
To allow Useres to know if their component selection for a Build is compatible or not through server-side compatibiliy checks

### Files Changed
- Created file, and wrote an endpoint that returns all Interfaces in new `InterfaceController.cs` controller
- Created file, and wrote fetch that gets all Interfaces in new `interfaceManager.js` manager
- Created initial Build DTO for bundling errors found during Build creation or updating to send back to the User in new `BuildErrorsDTO.cs` DTO class
- Added error and compatibility checking to Create and Update endpoints in `BuildController.cs`
- Moved fetch parsing logic from from createBuild to components to increase flexibility in `buildManager.js`
- Added logic and rendering to handle displaying errors found during the Build creation or update process in `BuildForm.jsx`
- Updated rendering or Memory Modules and Storage Devices to show their size in `BuildDetails.jsx`

### To Test
Fetch, setup, and run according to project README, then confirm the following
- Attempting to submit a new or edited build without a selection for each component renders errors for each missing component
- Attempting to submit a new or edited build with components that are not compatible with each other renders errors for each incompatible component
- Attempting to submit a new or edited build with more memory modules or storage devices than your motherboard supports will render errors that say as such
- Attempting to submit a new or edited build with a PSU that cannot provide enough wattage for the selected components will render an error that states as such
- Attempting to submit a new or edited build with a CPU Cooler than cannot dissipate as much heat as the CPU can produce will render an error that states as such
- User can only submit a new or edited Build that has no errors
- Memory and Storage show their size in the Build Details view

closes #33 
closes #34 
closes #35 
closes #36 
closes #37
